lib/archethic/metrics/generalise_metric_structure.ex

```diff
@@ -0,0 +1,156 @@
+defmodule ArchEthic.Metrics.GeneraliseMetricStructure do
+  @moduledoc """
+  MetricsDataParser.run() parses metrics data and convert into structured data
+  """
+
+  def run(content) do
```

Comment:
you can provide also a typespec to help dialyzer catches some errors.
As well you can include doctest to ensure the parsing works as expected